cosmos/lib/cosmos/packets/structure.rb

```diff
@@ -525,15 +526,11 @@ def synchronize_allow_reads(top = false)
     module MethodMissing
       # Method missing provides reading/writing item values as if they were methods to the class
       def method_missing(name, value = nil)
-        if @buffer
-          if value
-            # Strip off the equals sign before looking up the item
-            return write(name.to_s[0..-2], value)
-          else
-            return read(name.to_s)
-          end
+        if value
+          # Strip off the equals sign before looking up the item
+          return write(name.to_s[0..-2], value)
         else
-          raise "No buffer available for method_missing"
+          return read(name.to_s)
```

Comment:
So we always have @buffer now?

Reply:
Effectively.  It will allocate if needed.